### ISSUE TITLE
Suggestion to implement single-line asterisks

### ISSUE THREAD

**Issue by ftolsson** [lmparppei/Beat#106]
Opened this as an issue to be able to collect what I *think* is the solution to getting single-line asterisks through CSS.

Changes necessary to the Beat HTML output:

* instead of (or in addition to) giving the entire element a class of `.changed`, 
* wrap every individual range of changed text in `<span>` (tentatively called `.changedDetail` here)
* at the start of every span, add an empty `<a></a>` element (tentatively called `.starBox`here)

So I bet you'll come up with better class names, but that would give an output looking something like this:

```html
This is a text with some <span class="changedDetail"><a class="starBox"></a>changed words</span> in it.
```

The CSS needed for this to work would then be: 

```css
.changedDetail {
    position: relative !important;
    background-color: rgba(255, 128, 0, 0.3);    /*only for showing the changed text here*/
    display: inline;
    height: auto;
}

.starbox{
    float: left;
    width: 100%;
    left: initial;
    height: inherit;
}

.starbox::after {
    position: absolute;
    top: 0px;
    left: initial;
      
    width: 8px !important;
    height: calc(100% - 4px);

    content: '********************';
    overflow: hidden !important;
    font-weight: normal;
    text-align: right !important;
    letter-spacing: -1pt;

    z-index: 999;

    transform: translateX(calc(var(--action-width) + 20px)) !important;
}

.dialogue .starbox::after{
    transform: translateX(calc(var(--action-width) - var(--dialogue-indent) + 20px)) !important;
}

.character .starbox::after{
    transform: translateX(calc(var(--action-width) - var(--character-indent) + 20px)) !important;
}

.parenthetical .starbox::after{
    transform: translateX(calc(var(--action-width) - var(--parenthetical-indent) + 20px)) !important;
}

/* and so on for all element types*/
```

The variables are already in use in the current CSS for setting widths and indents to elements, so it all follows a very easy and already present logic. The CSS should be fixable in a matter of minutes, the hard part is whether you are easily able to add those extra html tags to the output. 

But bottomline, to me this seems like a quite failsafe way to use CSS for asterisks on changed lines only!

<img width="653" alt="image" src="https://user-images.githubusercontent.com/25481285/153270020-83b85e43-ba16-4562-a509-24691db90aa6.png">

**Comment by ftolsson** [comment]
Isn't this one just about ready to be closed? :)

**Comment by lmparppei** [comment]
Not yet, I haven't put your CSS in and there are some other things I need to sort out. So let's keep it open for further discussion until it's actually finished!

**Comment by lmparppei** [comment]
Implemented in `1.93.1`
